plugins/modules/sonic_static_routes.py

```diff
@@ -0,0 +1,241 @@
+#
+# Changes should be made in the model used to
+#   generate this file or in the resource module
+#   builder template.
+#
+#############################################
+
+"""
+The module file for sonic_static_routes
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = """
+---
+module: sonic_static_routes
+version_added: 2.0.0
+short_description: Manage static routes configuration on SONiC
+description:
+  - This module provides configuration management of static routes for devices running SONiC
+author: "Shade Talabi (@stalabi1)"
+options:
+  config:
+    type: list
+    elements: dict
+    description:
+      - Manages 'static_routes' configurations
+    suboptions:
+      vrf_name:
+        required: True
+        type: str
+        description:
+          - Name of the configured VRF on the device.
+      static_list:
+        type: list
+        elements: dict
+        description:
+          - A list of 'static_routes' configurations.
+        suboptions:
+          prefix:
+            required: True
+            type: str
+            description:
+              - List of locally configured static routes.
+          next_hops:
+            type: list
+            elements: dict
+            description:
+              - A list of next-hops to be utilised for the static route being specified.
+            suboptions:
+              index:
+                required: True
+                type: dict
+                description:
+                  - An identifier utilised to uniquely reference the next-hop.
+                suboptions:
+                  blackhole:
+                    type: bool
+                    default: False
+                    description:
+                      - Indicates if route is blackholed.
+                  interface:
+                    type: str
+                    description:
+                      - The reference to a base interface.
+                  network_instance:
+                    type: str
+                    description:
+                      - Name of the next-hop network instance for leaked routes.
+                  next_hop:
+                    type: str
+                    description:
+                      - The next-hop that is to be used for the static route.
+              metric:
+                type: int
+                description:
+                  - Specifies the preference of the next-hop entry when it is injected into the RIB.
+              track:
+                type: int
+                description:
+                  - The IP SLA track ID for static route.
+              tag:
+                type: int
+                description:
+                  - The tag value for the static route.
+  state:
+    description:
+      - The state of the configuration after module completion.
+    type: str
+    choices:
+    - merged
+    - deleted
+    default: merged
+"""
+EXAMPLES = """
```

Comment:
It would be helpful to add a "blackhole" next-hop example, and a merged/modify case that changes an attribute.

The "delete" case would be more realistic if it deleted a single next-hop for the target route (instead of all of them, which is sort of an invalid operation).